x-pack/plugins/security/server/routes/api/public/roles/get.js

```diff
@@ -15,6 +15,34 @@ export function initGetRolesApi(server, callWithRequest, routePreCheckLicenseFn,
     const roleKibanaApplications = roleApplications
       .filter(roleApplication => roleApplication.application === application);
 
+    // if any application entry contains an empty resource, we throw an error
+    if (roleKibanaApplications.some(entry => entry.resources.length === 0)) {
+      throw new Error(`ES returned an application entry without resources, can't process this`);
```

Comment:
Is there a reason we can't return `{ success: false }` for this case?

Reply:
This really shouldn't ever happen, if it does Elasticsearch's validation rules have changed drastically. I chose to throw an Error explicitly here because the contract with ES has changed if this does occur, so it really is an unanticipated situation. The other transform errors are used when a user reaches around our APIs and writes directly to ES, so that's why these two things are being handled differently.

This error used to be being thrown below where we'd try to parse a feature incorrectly and the serializer would throw the error for us, but with the changes that I've made to support the reserved privileges, it was no longer hitting this flow so we're throwing this error explicitly.

Reply:
I that makes sense, thanks for explaining!